numpydoc/numpydoc.py

```diff
@@ -184,6 +183,31 @@ class NumpyCDomain(ManglingDomainBase, CDomain):
     }
 
 
+def match_items(lines, content_old):
+    """Create the right items for lines.
+
+    This tries to recover where a line in ``lines``
+    came from before mangling.
+
+    It assumes that missing or new lines
+    are always empty.
```

Comment:
the docstring of this method? Sure.
And yeah, this function is certainly wrong. However, it is correct in most cases. Doing this properly would mean keeping track of all changes that happen in numpydoc, which is a major undertaking, I think.